sdk/core/azure-core/src/main/java/com/azure/core/http/rest/PagedFlux.java

```diff
@@ -63,4 +66,28 @@ public PagedFlux(Supplier<Mono<PagedResponse<T>>> firstPageRetriever,
                      Function<String, Mono<PagedResponse<T>>> nextPageRetriever) {
         super(firstPageRetriever, nextPageRetriever);
     }
+
+    /**
+     * Maps this PagedFlux instance of T to a PagedFlux instance of type S as per the provided mapper function.
+     *
+     * @param mapper The mapper function to convert from type T to type S.
+     * @param <S> The mapped type.
+     * @return A PagedFlux of type S.
+     */
+    public <S> PagedFlux<S> mapTo(Function<T, S> mapper) {
+        return new PagedFlux<S>(() -> getFirstPageRetriever().get()
+            .map(pagedResponse -> new PagedResponseBase<HttpRequest, S>(pagedResponse.getRequest(),
+                pagedResponse.getStatusCode(),
+                pagedResponse.getHeaders(),
+                pagedResponse.getValue().stream().map(mapper).collect(Collectors.toList()),
+                pagedResponse.getNextLink(),
+                null)),
+            continuationToken -> getNextPageRetriever().apply(continuationToken)
+                .map(pagedResponse -> new PagedResponseBase<HttpRequest, S>(pagedResponse.getRequest(),
+                    pagedResponse.getStatusCode(),
+                    pagedResponse.getHeaders(),
+                    pagedResponse.getValue().stream().map(mapper).collect(Collectors.toList()),
+                    pagedResponse.getNextLink(),
+                    null)));
```

Comment:
You should consider extracting the function out of the map call so that you can reuse the same code without duplication.

Reply:
Done.

Reply:
Done